File: src/agents/Flutter/mantra.py
from src.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class MantraPersona(BaseActivePersona):
    """
    Core: PhD in Structural Integrity & Code Purity (Flutter) 🧘
    Especialista em Clean Architecture, desacoplamento de UI e padrões Dart.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Mantra", "🧘", "PhD Quality Architect", "Flutter"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Pureza do Código Flutter...")
        
        audit_rules = [
            {'regex': r"setState\(", 'issue': 'Acoplamento: Uso excessivo de setState. Considere um State Manager (Bloc/Provider).', 'severity': 'medium'},
            {'regex': r"catch\s*\(.*?\)\s*\{\s*\}", 'issue': 'Anti-padrão: Captura de exceção vazia (Bare except).', 'severity': 'critical'}
        ]
        
        results = self.find_patterns(('.dart',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        if "setState" in content:
            return f"Entropia Lógica: O objetivo '{objective}' exige modularidade. Em '{file}', o acoplamento de estado na UI dificulta a evolução da 'Orquestração de Inteligência Artificial'."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em arquitetura e pureza Flutter."

File: src_local/agents/Flutter/Content/mantra.ts
/**
 * 🧘 Mantra - PhD in Structural Integrity & Code Purity (Flutter)
 * Especialista em integridade estrutural, gestão de estado limpa e padrões de pureza Dart.
 */
import type { AuditFinding, AuditRule, StrategicFinding } from "../../base.ts";
import { BaseActivePersona } from "../../base.ts";

export class MantraPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Mantra";
        this.emoji = "🧘";
        this.role = "PhD Quality Architect";
        this.phd_identity = "Structural Integrity & Code Purity (Flutter)";
        this.stack = "Flutter";
    }

    public override async execute(context: any): Promise<any> {
        this.setContext(context);
        const findings = await this.performAudit();
        if (this.hub) {
            const qualityNodes = await this.hub.queryKnowledgeGraph("setState", "medium");
            const reasoning = await this.hub.reason(`Analyze the structural purity of a Flutter system with ${qualityNodes.length} setState/exception patterns. Recommend Bloc/Provider migration.`);
            findings.push({ file: "Quality Audit", agent: this.name, role: this.role, emoji: this.emoji, issue: `Sovereign Mantra: Pureza Flutter validada via Rust Hub. PhD Analysis: ${reasoning}`, severity: "INFO", stack: this.stack, evidence: "Knowledge Graph Quality Audit", match_count: 1 } as any);
        }
        return findings;
    }

    public override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: [".dart"],
            rules: [
                { regex: /setState\(/, issue: "Acoplamento: Uso excessivo de setState. Considere um State Manager (Bloc/Provider).", severity: "medium" },
                { regex: /catch\s*\(.*?\)\s*\{\s*\}/, issue: "Anti-padrão: Captura de exceção vazia detectada.", severity: "critical" }
            ]
        };
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const { extensions, rules } = this.getAuditRules();
        const results = await this.findPatterns(extensions, rules);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(_objective: string, file: string, content: string): StrategicFinding | string | null {
        if (content.includes("setState")) {
            return {
                file,
                issue: "Entropia Lógica: Acoplamento de estado na UI detectado. Isso viola o Mantra de Pureza Estrutural.",
                severity: "STRATEGIC",
                context: this.name
            };
        }
        return "PhD Purity: Analisando integridade estrutural da 'Higiene de Código'. Focando em desacoplamento de estado e robustez de exceções.";
    }

    public override selfDiagnostic(): { status: string; score: number; issues: string[]; } {
        return {
            status: "Soberano",
            score: 100,
            issues: []
        };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, PhD em Integridade Estrutural e Pureza de Código Flutter.`;
    }
}

